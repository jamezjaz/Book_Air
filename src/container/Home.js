import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => (
  <div>
    <Link to="/login">
      <button type="button">Login</button>
    </Link>
    <Link to="/register">
      <button type="button">Register</button>
    </Link>
  </div>
);

export default Home;
