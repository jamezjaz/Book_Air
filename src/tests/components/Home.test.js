import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../../container/Home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Home />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it contains a button element', () => {
    const wrapper = shallow(<Home />);
    const button = <button type="button">Login</button>;
    expect(wrapper.containsMatchingElement(button)).toBe(true);
  });

  test('renders text in the dom correctly', () => {
    render(
      <Router>
        <Home />
      </Router>,
    );
    expect(screen.getByText('Login')).toBeInTheDocument();
  });
});
