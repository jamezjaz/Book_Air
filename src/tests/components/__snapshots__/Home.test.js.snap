// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div>
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Login
    </button>
  </a>
  <a
    href="/register"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Register
    </button>
  </a>
</div>
`;
